paasta_tools/autoscaling/autoscaling_service_lib.py

```diff
@@ -453,38 +519,43 @@ def autoscale_services(soa_dir=DEFAULT_SOA_DIR):
             cluster = load_system_paasta_config().get_cluster()
             configs = get_configs_of_services_to_scale(cluster=cluster, soa_dir=soa_dir)
             if configs:
-                marathon_config = load_marathon_config()
-                marathon_client = get_marathon_client(
-                    url=marathon_config.get_url(),
-                    user=marathon_config.get_username(),
-                    passwd=marathon_config.get_password())
-                all_marathon_tasks = marathon_client.list_tasks()
-                all_mesos_tasks = get_all_running_tasks()
                 with ZookeeperPool():
                     for config in configs:
                         try:
-                            job_id = config.format_marathon_app_dict()['id']
-                            # Get a dict of healthy tasks, we assume tasks with no healthcheck defined
-                            # are healthy. We assume tasks with no healthcheck results but a defined
-                            # healthcheck to be unhealthy (unless they are "old" in which case we
-                            # assume that marathon has screwed up and stopped healthchecking but that
-                            # they are healthy
-                            log.info("Inspecting %s for autoscaling" % job_id)
-                            marathon_tasks = {task.id: task for task in all_marathon_tasks
-                                              if task.id.startswith(job_id) and
-                                              (is_task_healthy(task) or not
-                                               marathon_client.get_app(task.app_id).health_checks or
-                                               is_old_task_missing_healthchecks(task, marathon_client))}
-                            if not marathon_tasks:
-                                raise MetricsProviderNoDataError("Couldn't find any healthy marathon tasks")
-                            mesos_tasks = [task for task in all_mesos_tasks if task['id'] in marathon_tasks]
+                            marathon_tasks, mesos_tasks = get_autoscaling_tasks(config)
                             autoscale_marathon_instance(config, list(marathon_tasks.values()), mesos_tasks)
                         except Exception as e:
                             write_to_log(config=config, line='Caught Exception %s' % e)
     except LockHeldException:
         log.warning("Skipping autoscaling run for services because the lock is held")
 
 
+def get_autoscaling_tasks(config):
+    marathon_config = load_marathon_config()
+    marathon_client = get_marathon_client(
```

Comment:
If @oktopuz's change is merged, this might significantly slow down paasta status again because it will be a new client for every instance and not be cached.